buildSrc/src/main/java/com/uber/okbuck/core/util/ProjectUtil.java

```diff
@@ -88,4 +100,59 @@ static String findVersionInClasspath(Project project, String group, String modul
                 .map(ModuleComponentIdentifier::getVersion)
                 .orElse(null);
     }
+
+    /*
+     * Copyright (C) 2017 The Android Open Source Project
+     *
+     * Licensed under the Apache License, Version 2.0 (the "License");
+     * you may not use this file except in compliance with the License.
+     * You may obtain a copy of the License at
+     *
+     *      http://www.apache.org/licenses/LICENSE-2.0
+     *
+     * Unless required by applicable law or agreed to in writing, software
+     * distributed under the License is distributed on an "AS IS" BASIS,
+     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     * See the License for the specific language governing permissions and
+     * limitations under the License.
+     */
+    // Copied from AGP 3.1.0 ArtifactDependencyGraph
+    public static void downloadSources(
+            Project project,
+            Set<ComponentIdentifier> artifacts) {
+        final DependencyHandler dependencies = project.getDependencies();
+
+        try {
+            ArtifactResolutionQuery query = dependencies.createArtifactResolutionQuery();
+            query.forComponents(artifacts);
+
+            @SuppressWarnings("unchecked")
+            Class<? extends Artifact>[] artifactTypesArray =
+                    (Class<? extends Artifact>[]) new Class<?>[] {SourcesArtifact.class};
+            query.withArtifacts(JvmLibrary.class, artifactTypesArray);
+            ArtifactResolutionResult queryResult = query.execute();
+            Set<ComponentArtifactsResult> resolvedComponents = queryResult.getResolvedComponents();
+
+            // Create and execute another query to attempt javadoc resolution
+            // where sources are not available
+            Set<ComponentIdentifier> remainingToResolve = Sets.newHashSet();
+            for (ComponentArtifactsResult componentResult : resolvedComponents) {
+                Set<ArtifactResult> sourcesArtifacts =
+                        componentResult.getArtifacts(SourcesArtifact.class);
+                if (sourcesArtifacts.isEmpty()) {
+                    remainingToResolve.add(componentResult.getId());
+                }
+            }
+            if (!remainingToResolve.isEmpty()) {
```

Comment:
we dont care about javadoc artifacts, so lets remove this logic. We only care about sources

Reply:
done